presto-main/src/test/java/io/prestosql/sql/analyzer/TestTypeSignatureTranslator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.analyzer;
+
+import io.prestosql.sql.parser.SqlParser;
+import org.testng.annotations.Test;
+
+import static io.prestosql.sql.analyzer.TypeSignatureTranslator.toDataType;
+import static io.prestosql.sql.analyzer.TypeSignatureTranslator.toTypeSignature;
+import static java.util.Locale.ENGLISH;
+import static org.testng.Assert.assertEquals;
+
+public class TestTypeSignatureTranslator
+{
+    private static final SqlParser SQL_PARSER = new SqlParser();
+
+    private void assertRoundTrip(String expression)
+    {
+        // TODO: remove this toLowerCase to match standard SQL semantics
+        assertEquals(SQL_PARSER.createType(expression.toLowerCase(ENGLISH)), toDataType(toTypeSignature(SQL_PARSER.createType(expression))));
+    }
+
+    @Test
+    public void testSimpleTypes()
+    {
+        assertRoundTrip("VARCHAR");
```

Comment:
I think we also need parameterized tests for `varchar`, `timestamp`, `time`, `decimal`.

Reply:
Have added for `varchar` and `decimal` but since we don't support parameterized `time` and `timestamp` in Presto I guess it is not possible to add for those two types.